java/flight/src/main/java/org/apache/arrow/flight/ArrowMessage.java

```diff
@@ -309,7 +311,7 @@ private InputStream asInputStream(BufferAllocator allocator) {
         size += b.readableBytes();
         // [ARROW-4213] These buffers must be aligned to an 8-byte boundary in order to be readable from C++.
         if (b.readableBytes() % 8 != 0) {
-          int paddingBytes = 8 - (b.readableBytes() % 8);
+          int paddingBytes = checkedCastToInt(8 - (b.readableBytes() % 8));
```

Comment:
This is a little nit-picking: 
checkedCastToInt should be applied to b.readableBytes(), so the following arithmetic will not be promoted to int64?

The performance impact is trivial, so you can choose not to change it.

Reply:
I think it actually shouldn't be applied at all since the value is guaranteed to be less then 8 after the mod, so I've changed this to a hard cast.